lib/common/service-object.js

```diff
@@ -0,0 +1,303 @@
+ * @param {object} callback.apiResponse - The full API response.
+ */
+ServiceObject.prototype.create = function(options, callback) {
+  var self = this;
+  var args = [this.id];
+
+  if (is.fn(options)) {
+    callback = options;
+  }
+
+  if (is.object(options)) {
+    args.push(options);
+  }
+
+  // Wrap the callback to return *this* instance of the object, not the newly-
+  // created one.
+  function onCreate(err, instance, apiResponse) {
+    if (err) {
+      callback(err, null, apiResponse);
+      return;
+    }
+
+    self.metadata = instance.metadata;
+
+    callback(null, self, apiResponse);
+  }
+
+  args.push(onCreate);
+
+  this.createMethod.apply(null, args);
+};
+
+/**
+ * Delete the object.
+ *
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.apiResponse - The full API response.
+ */
+ServiceObject.prototype.delete = function(callback) {
+  var reqOpts = {
+    method: 'DELETE',
+    uri: ''
+  };
+
+  // The `request` method may have been overridden to hold any special behavior.
+  // Ensure we call the original `request` method.
+  ServiceObject.prototype.request.call(this, reqOpts, function(err, resp) {
+    callback(err, resp);
+  });
+};
+
+/**
+ * Check if the object exists.
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {boolean} callback.exists - Whether the object exists or not.
+ */
+ServiceObject.prototype.exists = function(callback) {
+  this.get(function(err) {
+    if (err) {
+      if (err.code === 404) {
+        callback(null, false);
+      } else {
+        callback(err);
+      }
+
+      return;
+    }
+
+    callback(null, false);
+  });
+};
+
+/**
+ * Get the object if it exists. Optionally have the object created if an options
+ * object is provided with `autoCreate: true`.
+ *
+ * @param {object=} config - The configuration object that will be used to
+ *     create the object if necessary.
+ * @param {boolean} config.autoCreate - Create the object if it doesn't already
+ *     exist.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.instance - The instance.
+ * @param {object} callback.apiResponse - The full API response.
+ */
+ServiceObject.prototype.get = function(config, callback) {
+  var self = this;
+
+  if (is.fn(config)) {
+    callback = config;
+    config = {};
+  }
+
+  config = config || {};
+
+  var autoCreate = config.autoCreate && is.fn(this.create);
+  delete config.autoCreate;
+
+  this.getMetadata(function(err, metadata) {
+    if (err) {
+      if (err.code === 404 && autoCreate) {
+        if (!is.empty(config)) {
+          self.create(config, callback);
+        } else {
+          self.create(callback);
```

Comment:
Warning: nitpick ahead! Feel free to ignore :)

What are your feelings on something like..

```javascript
var args = [callback];

if (!is.empty(config)) {
  args.unshift(config);
}

self.create.apply(self, args);
```

?